diff --git a/externals/mrpeach/net/tcpserver.c b/externals/mrpeach/net/tcpserver.c
index 0a0778857..3710ea986 100644
--- a/externals/mrpeach/net/tcpserver.c
+++ b/externals/mrpeach/net/tcpserver.c
@@ -508,7 +508,7 @@ failed:
 /* SIOCOUTQ exists only(?) on linux, returns remaining space in the socket's output buffer  */
 static int tcpserver_send_buffer_avaliable_for_client(t_tcpserver *x, int client)
 {
-    int sockfd = x->x_sr[client].sr_fd;
+    int sockfd = x->x_sr[client]->sr_fd;
     int result = 0L;
 
     ioctl(sockfd, SIOCOUTQ, &result);
@@ -899,7 +899,7 @@ static void *tcpserver_broadcast_thread(void *arg)
                     {
                         /* this might be better than allocating huge amounts of memory */
 #ifdef SIOCOUTQ
-                        if (tcpserver_send_buffer_avaliable_for_client(x, client) < j)
+                        if (tcpserver_send_buffer_avaliable_for_client(ttbp->x, client) < j)
                         {
                             error("%s_broadcast_thread: buffer too small for client(%d)", objName, client);
                             freebytes (arg, sizeof (t_tcpserver_broadcast_params));
@@ -942,7 +942,7 @@ static void *tcpserver_broadcast_thread(void *arg)
         for(client = 0; client < ttbp->x->x_nconnections; client++)	/* check if connection exists */
         {
 #ifdef SIOCOUTQ
-            if (tcpserver_send_buffer_avaliable_for_client(x, client) < length)
+            if (tcpserver_send_buffer_avaliable_for_client(ttbp->x, client) < length)
             {
                 error("%s_broadcast_thread: buffer too small for client(%d)", objName, client);
                 freebytes (arg, sizeof (t_tcpserver_broadcast_params));
